geode-core/src/main/java/org/apache/geode/internal/cache/execute/FunctionContextImpl.java

```diff
@@ -37,20 +38,25 @@
 
   private String functionId = null;
 
+  private Cache cache = null;
+
   private ResultSender resultSender = null;
 
   private final boolean isPossDup;
 
   public FunctionContextImpl(final String functionId, final Object args,
       ResultSender resultSender) {
-    this.functionId = functionId;
-    this.args = args;
-    this.resultSender = resultSender;
-    this.isPossDup = false;
+    this(null, functionId, args, resultSender, false);
+  }
+
+  public FunctionContextImpl(final Cache cache, final String functionId, final Object args,
+      ResultSender resultSender) {
+    this(cache, functionId, args, resultSender, false);
   }
 
-  public FunctionContextImpl(final String functionId, final Object args, ResultSender resultSender,
-      boolean isPossibleDuplicate) {
+  public FunctionContextImpl(final Cache cache, final String functionId, final Object args,
+      ResultSender resultSender, boolean isPossibleDuplicate) {
```

Comment:
this could still have have cache end up being null if user did not create the FunctionContextImpl with a cache, either pass in null, or use the other non-cache constructor. It would be good to add a check here:
if(cache==null){
  cache = GemfireCacheImpl.getExisting();
}